src/events.rs

```diff
@@ -159,6 +162,101 @@ pub struct Event {
     pub sig: String,
 }
 
+#[derive(Error, Debug, Eq, PartialEq)]
+pub enum EventError {
+    #[error("Secp256k1 Error: {}", _0)]
+    Secp256k1Error(secp256k1::Error),
+}
+
+impl From<secp256k1::Error> for EventError {
+    fn from(err: secp256k1::Error) -> Self {
+        Self::Secp256k1Error(err)
+    }
+}
+
+impl Event {
+    /// get_content returns the content of the event
+    /// # Example
+    /// ```rust
+    /// use nostr_rust::{events::EventPrepare, utils::get_timestamp, Identity};
+    /// use std::str::FromStr;
+    ///
+    /// let actual_time = get_timestamp();
+    /// let identity = Identity::from_str(env!("SECRET_KEY")).unwrap();
+    /// let event = EventPrepare {
+    ///    pub_key: env!("PUBLIC_KEY").to_string(),
+    ///    created_at: get_timestamp(),
+    ///    kind: 0,
+    ///    tags: vec![],
+    ///    content: "content".to_string(),
+    /// }.to_event(&identity, 0);
+    ///
+    /// assert_eq!(event.get_content(), format!("[0,\"c5aec31e83bdf980939b5ef7c6bcaa2be8bd39d38667da58ba6dba240eb8b69d\",{},0,[],\"content\"]", actual_time));
+    /// ```
+    pub fn get_content(&self) -> String {
+        json!([
+            0,
+            self.pub_key,
+            self.created_at,
+            self.kind,
+            self.tags,
+            self.content
+        ])
+        .to_string()
+    }
+
+    /// Get the id of the event which is the sha256 hash of the content
+    /// # Example
+    /// ```rust
+    /// use nostr_rust::{events::EventPrepare, Identity};
+    /// use std::str::FromStr;
+    /// let identity = Identity::from_str(env!("SECRET_KEY")).unwrap();
+    /// let event = EventPrepare {
+    ///   pub_key: env!("PUBLIC_KEY").to_string(),
+    ///   created_at: 0, // Don't use this in production
+    ///   kind: 0,
+    ///   tags: vec![],
+    ///   content: "content".to_string(),
+    /// }.to_event(&identity, 0);
+    ///
+    /// assert_eq!(event.get_content_id(), "4a57aad22fc0fd374e8ceeaaaf8817fa6cb661ca2229c66309d7dba69dfe2359");
```

Comment:
Same here, except it's going to be hard to test, so I think we can test the length of the id first.
I can't think of a better way to simplify the testing here.

Reply:
I used the same content id as the one in the `EventPrepare` tests. So it should pass with the key pair that the CI uses. But changing it to length makes testing locally work.

Reply:
Will change this